creator-node/test/CNodeHealthManager.test.js

```diff
@@ -0,0 +1,478 @@
+    const dailySyncFailCount = 9
+    const verboseHealthCheckResp = {
+      dailySyncSuccessCount,
+      dailySyncFailCount
+    }
+    expect(() => determinePeerHealth(verboseHealthCheckResp)).to.throw(
+      `Latest daily sync data shows that this node fails at a high rate of syncs. Successful syncs=${dailySyncSuccessCount} || Failed syncs=${dailySyncFailCount}. Minimum successful sync percentage=${minimumSuccessfulSyncCountPercentage}%`
+    )
+  })
+
+  it('throws when historical sync success rate for rolling 30-day window is below threshold', function () {
+    const thirtyDayRollingSyncSuccessCount = 1
+    const thirtyDayRollingSyncFailCount = 9
+    const verboseHealthCheckResp = {
+      thirtyDayRollingSyncSuccessCount,
+      thirtyDayRollingSyncFailCount
+    }
+    expect(() => determinePeerHealth(verboseHealthCheckResp)).to.throw(
+      `Rolling sync data shows that this node fails at a high rate of syncs. Successful syncs=${thirtyDayRollingSyncSuccessCount} || Failed syncs=${thirtyDayRollingSyncFailCount}. Minimum successful sync percentage=${minimumSuccessfulSyncCountPercentage}%`
+    )
+  })
+})
+
+describe('test CNodeHealthManager -- isPrimaryHealthy()', function () {
+  let sandbox
+  beforeEach(function () {
+    sandbox = sinon.createSandbox()
+  })
+
+  afterEach(function () {
+    sandbox.restore()
+  })
+
+  const primary = 'http://cn1.co'
+  const gracePeriodSeconds = 1
+  config.set('maxNumberSecondsPrimaryRemainsUnhealthy', 1)
+
+  it('returns true (healthy) when healthy', async function () {
+    // Make isNodeHealthy return true
+    const isNodeHealthyStub = sandbox
+      .stub(CNodeHealthManager, 'isNodeHealthy')
+      .resolves(true)
+
+    const isHealthy = await CNodeHealthManager.isPrimaryHealthy(primary)
+    expect(isHealthy).to.be.true
+    expect(isNodeHealthyStub).to.have.been.calledOnceWithExactly(primary, true)
+  })
+
+  it('returns true when health check fails during grace period, then false when grace period ends, then true when health check starts passing again', async function () {
+    // Mock CNodeHealthManager to use the config with our shorter grace period
+    const CNodeHealthManagerMock = proxyquire(
+      '../src/services/stateMachineManager/CNodeHealthManager.js',
+      {
+        './../../config': config
+      }
+    )
+    // Make isNodeHealthy return false
+    const isNodeHealthyStub = sandbox
+      .stub(CNodeHealthManagerMock, 'isNodeHealthy')
+      .resolves(false)
+
+    // Verify that the node is marked as healthy during the grace period (even though it's unhealthy)
+    const isHealthy = await CNodeHealthManagerMock.isPrimaryHealthy(primary)
+    expect(isHealthy).to.be.true
+    const isHealthyDuringGracePeriod =
+      await CNodeHealthManagerMock.isPrimaryHealthy(primary)
+    expect(isHealthyDuringGracePeriod).to.be.true
+
+    // Verify that the node is unhealthy after the grace period ends
+    await Utils.timeout(gracePeriodSeconds * 1000 + 1)
```

Comment:
I think its generally better practice to not stall on a test, even if the test requires it. you did the right thing, which was set the threshold super low. the test threshold is I think about ~30s each anyway. but, let's try to avoid waiting in the future 👍